src/main/java/org/csanchez/jenkins/plugins/kubernetes/pipeline/ContainerExecDecorator.java

```diff
@@ -244,23 +244,22 @@ public void kill(Map<String, String> modelEnvVars) throws IOException, Interrupt
             }
 
 
-            private boolean isContainerReady(Pod pod, String container) {
-                if (pod == null || pod.getStatus() == null || pod.getStatus().getContainerStatuses() == null) {
-                    return false;
-                }
-
-                for (ContainerStatus info : pod.getStatus().getContainerStatuses()) {
-                    if (info.getName().equals(container) && info.getReady()) {
-                        return true;
-                    }
-                }
-                return false;
-            }
-
             private void waitUntilContainerIsReady() throws IOException {
                 try {
-                    client.pods().inNamespace(namespace).withName(podName)
+                    Pod pod = client.pods().inNamespace(namespace).withName(podName)
                             .waitUntilReady(CONTAINER_READY_TIMEOUT, TimeUnit.MINUTES);
+
+                    for (ContainerStatus info : pod.getStatus().getContainerStatuses()) {
```

Comment:
there was a check for `pod.getStatus() == null || pod.getStatus().getContainerStatuses() == null` IIRC null was possible during startup

Reply:
I thought that `waitUntilReady` should take care of that, but I'll re-add it to make sure.